modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2407,10 +2421,18 @@ private DynamicObjectDefinitionTable _getDynamicObjectDefinitionTable(
 		ObjectDefinition objectDefinition =
 			_objectDefinitionPersistence.findByPrimaryKey(objectDefinitionId);
 
-		return new DynamicObjectDefinitionTable(
-			objectDefinition,
+		List<ObjectField> objectFieldList =
 			_objectFieldLocalService.getObjectFields(
-				objectDefinitionId, objectDefinition.getDBTableName()),
+				objectDefinitionId, objectDefinition.getDBTableName());
+
+		if (_restrictedFields != null) {
+			objectFieldList.removeIf(
```

Comment:
We should confirm what the requirement is. 
I understood from the ticket that the front-end will pass the fields that should be in the response, not that should be excluded from it.
If the first alternative is the expected one, than I think we should rename the parameter to something along the lines of `selectedFields`.